### PR TITLE
fix genesis

### PR DESCRIPTION
**Pull request by ChihChengLiang**
```
make init
./build/hubble load ../contracts/genesis.json # loads genesis to the config.toml
./build/hubble start
```

### CODE REVIEW

File: cmd/start.go [hunk 1]
```diff
@@ -93,14 +95,26 @@ func startCmd() *cobra.Command {
 	}
 }
 
+type GenUserState struct {
```
Comment (ChihChengLiang) [inline cmd/start.go:98]: Had to keep this before we fix our tree initialization

**Review by ChihChengLiang — COMMENTED** [review]
I haven't successfully run the `/build/hubble start`, since the address are still loaded from the `config.toml`. Will fix that to complete this PR.

File: cmd/start.go [hunk 2]
```diff
@@ -88,27 +85,23 @@ func startCmd() *cobra.Command {
...
+	err := DBI.InitStateTree(int(cfg.MaxTreeDepth))
+	common.PanicIfError(err)
 
-	err = DBI.InitStateTree(depth)
+	opts := bind.CallOpts{From: ethCmn.HexToAddress(cfg.OperatorAddress)}
+	accountTreeDepth, err := bz.SC.AccountRegistry.DEPTH(&opts)
 	common.PanicIfError(err)
 
-	err = DBI.InitAccountTree(depth)
+	err = DBI.InitAccountTree(int(accountTreeDepth.Uint64()))
```
Comment (ChihChengLiang) [inline cmd/start.go:101]: AccountTree depth should come from the contract.

File: config/config.go [hunk 3]
```diff
@@ -59,9 +59,14 @@ type Configuration struct {
...
-	OperatorAddress   string `mapstructure:"operator_address"`
-	LastRecordedBlock string `mapstructure:"last_recorded_block"`
+	OperatorKey     string `mapstructure:"operator_key"`
+	OperatorAddress string `mapstructure:"operator_address"`
+
+	MaxTreeDepth      uint64 `mapstructure:"max_tree_depth"`
+	MaxDepositSubtree uint64 `mapstructure:"max_deposit_subtree"`
+	StakeAmount       uint64 `mapstructure:"stake_amount"`
+
+	GenesisEth1Block uint64 `mapstructure:"genesis_eth1_block"`
```
Comment (ChihChengLiang) [inline config/config.go:69]: LastRecordedBlock is replaced with GenesisEth1Block

File: cmd/init.go [hunk 4]
```diff
@@ -22,10 +22,37 @@ func initCmd() *cobra.Command {
...
+}
+
+func configureGenesisCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "load",
+		Short: "Load Genesis from a genesis.json",
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg, err := config.ParseConfig()
+			common.PanicIfError(err)
+			genesis, err := config.ReadGenesisFile(args[0])
```
Comment (vaibhavchellani) [inline cmd/init.go:36]: Do you think we should use the flags here? Like the other cli commands?

**Review by vaibhavchellani — APPROVED** [review]
Amazing work! LGTM

File: cmd/init.go (hunk 4 above)
Comment (ChihChengLiang) [reply cmd/init.go:36]: I was thinking about using the flag too. But then I realized the path is already a required argument and is the only argument. So we don't have to key in a flag in the command.

**Pull request merged**